drivers/net/ethernet/microchip/enc28j60.c

```diff
@@ -1076,12 +1076,14 @@ static void enc28j60_tx_clear(struct net_device *ndev, bool err)
 	else
 		ndev->stats.tx_packets++;
 
+	netif_tx_lock(ndev);
```

Comment:
Normally we should use the netif_tx_lock_bh() variant here, since the assignment of tx_skb in send_packet() happens in softirq context. However in this case we dont need the lock for exclusive access to tx_skb, since the netif queue steering already prevents this function from running concurrently with  send_packet() (thus we dont need to disable bottom halves).  We only need the lock for smp consistency: If send_packet() runs on processor A and assigns priv->tx_skb and afterwards  tx_clear runs on processor B and checks priv->tx_skb we must ensure that the change done in A is actually visible in the perception of B. This is done by locking, since locking implies the necessary memory barriers. But we can also do without locking and use an explicit SMB memory barrier instead:

smp_rmb(); /* make sure to read the up-to-date value of priv->tx_skb that has been set
                       by send_packet() */ 
if (priv->tx_skb)
...